File: python/broadcast.py
#!/usr/bin/env python
# File: broadcast.py
# Name: D.Saravanan
# Date: 07/03/2023

""" Script to broadcast data to all processes """

from mpi4py import MPI
from numpy import array


def main():
    """broadcast data to all processes"""
    comm = MPI.COMM_WORLD
    rank = comm.Get_rank()
    size = comm.Get_size()

    if rank == 0:
        data = {
            "key1": [10, 10.1, 10 + 11j],
            "key2": ["mpi4py", "python"],
            "key3": array([1, 2, 3]),
        }
    else:
        data = None

    # broadcast data to all processes
    data = comm.bcast(data, root=0)

    if rank == 0:
        print("bcast finished")

    print(f"data on rank {rank} of size {size} is {data}")
    comm.Barrier()


if __name__ == "__main__":
    main()

File: julia/broadcast.jl
#!/usr/bin/env julia
# File: broadcast.jl
# Name: D.Saravanan
# Date: 07/03/2023

""" Script to broadcast data to all processes """

import MPI
MPI.Init()

function main()
    """broadcast data to all processes"""
    comm = MPI.COMM_WORLD
    rank = MPI.Comm_rank(comm)
    size = MPI.Comm_size(comm)

    if rank == 0
        data = Dict(
            "key1" => [10, 10.1, 10 + 11im],
            "key2" => ["mpi-julia", "julia"],
            "key3" => Array([1, 2, 3]),
        )
    else
        data = 0
    end

    # broadcast data to all processes
    data = MPI.bcast(data, root = 0, comm)

    if rank == 0
        println("bcast finished")
    end

    println("data on rank $rank of size $size is $data")

    MPI.Barrier(comm)
    MPI.Finalize()
end

main()
